src/transaction/transaction_master_group_complete_manager.cpp

```diff
@@ -77,29 +78,45 @@ namespace cubtx
   //
   void master_group_complete_manager::notify_stream_ack (const cubstream::stream_position stream_pos)
   {
-    /* TODO - disable it temporary since it is not tested */
-    return;
-
     /* TODO - consider quorum. Consider multiple calls of same thread. */
     if (stream_pos >= m_latest_closed_group_end_stream_position)
       {
 	cubthread::entry *thread_p = &cubthread::get_entry ();
 	do_complete (thread_p);
+	er_log_debug (ARG_FILE_LINE, "master_group_complete_manage::notify_stream_ack pos=%llu\n", stream_pos);
       }
   }
 
+  //
+  // get_manager_type - get manager type.
+  //
+  int master_group_complete_manager::get_manager_type () const
+  {
+    return LOG_TRAN_COMPLETE_MANAGER_MASTER_NODE;
+  }
+
   //
   // on_register_transaction - on register transaction specific to master node.
   //
   void master_group_complete_manager::on_register_transaction ()
   {
-    /* This function is called after adding a transaction to the current group.
-     * Currently, we wakeup GC thread when first transaction is added into current group.
-     */
-    if (get_current_group ().get_container ().size () == 1)
+    /* This function is called after adding a transaction to the current group. */
+    assert (get_current_group ().get_container ().size () >= 1);
+
+#if defined (SERVER_MODE)
+    if (is_latest_closed_group_completed ())
```

Comment:
is it necessary to have lock on  m_group_mutex, while waking up complete daemon and stream senders ?

Reply:
No. But, with the actual design, it was the simple way. Probably, it may be improved, if we consider that may affect the performance.